lms/djangoapps/oauth2_handler/handlers.py

```diff
@@ -192,17 +193,21 @@ def find_courses(self, user, access_type, values=None):
 
     # pylint: disable=missing-docstring
     def _get_courses_with_access_type(self, user, access_type):
-
         # Check the application cache and update if not present. The application
         # cache is useful since there are calls to different endpoints in close
-        # succession, for example the id_token and user_info endpoins.
+        # succession, for example the id_token and user_info endpoints.
 
         key = '-'.join([str(self.__class__), str(user.id), access_type])
         course_ids = cache.get(key)
 
-        if course_ids is None:
-            course_ids = [unicode(course.id) for course in _get_all_courses()
-                          if has_access(user, access_type, course)]
+        if not course_ids:
+            if GlobalStaff().has_user(user):
+                # Global staff have access to all courses, so just return them all.
+                course_ids = [unicode(course.id) for course in _get_all_courses()]
```

Comment:
Note that the same filtering is done in the function that list the all courses in the courseware: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/branding/__init__.py#L9